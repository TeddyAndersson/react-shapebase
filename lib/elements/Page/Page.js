import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Element = (props) => {
  const {
    children,
    className,
    size,
  } = props;

  return (
    <main className={className}>
        {children}
    </main>
  );
};

Element.propTypes = {
  /**
    * Primary content.
    */
  children: PropTypes.node,

  /**
    * Additonal classes.
    */
  className: PropTypes.string,
};


/** Styling the Component with styled-components */
const Page = styled(Element)`
    font-family: Helvetica;
    min-width: 100%;
    min-height: 100%;
    background-size: 300% 300%;
    background: #fff;

    ${props => props.color && css`
        background: ${props.color};
    `}
`;


export default Page;