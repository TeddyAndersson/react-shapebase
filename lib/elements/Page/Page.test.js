import React from 'react';
import Page from './Page'
import renderer from 'react-test-renderer';

test('should render Page', () => {
  const component = renderer.create(
    <Page>
      This is a Page component. It purpose is to cover the with and the hight of a Html Body.
    </Page>);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
