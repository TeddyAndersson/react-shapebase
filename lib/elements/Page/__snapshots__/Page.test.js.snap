// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page 1`] = `
<main
  className="sc-bdVaJa epNqie"
>
  This is a Page component. It purpose is to cover the with and the hight of a Html Body.
</main>
`;
